### DISCUSSION TITLE
How to get model pred_scores？

### DISCUSSION THREAD

**Discussion by ikun0124** [open-edge-platform/anomalib#819] [Q&A]
I want to get model output, such as the prediction score of the model. When I run the python test.py --config xx,  I only get the metric of model, but not the prediction of dataset.
I also try other method that load trained dataset and mode to predict output. But there are always bug in the sklearn.metric module. Besides I don't know how to call adaptive score calculation on non PL framework code.
I had to turn to the authors to solve the above problems.

**Comment by ntdgo** [comment]
Hi, try to get the output of `trainer.test` command in `test.py` script
https://github.com/openvinotoolkit/anomalib/blob/c82ddcd76bed47d196cffbee333a995e7f1df3ed/tools/test.py#L49

It should be return some information might you need: 
https://github.com/Lightning-AI/lightning/blob/ca88f813a440bef61e611dc3c40343a2774cd21a/src/pytorch_lightning/trainer/trainer.py#L769-L773

**Reply by ikun0124** [reply]
Hi, I have seen the link you gave me. It says that test can only return metric result, but I need the prediction of the model.

**Comment by ikun0124** [comment]
@thanh-repo I have seen the link you gave me. It says that test can only return metric result, but I need the prediction of the model.

**Comment by ashwinvaidya17** [comment]
@loyal-ikun have you looked at the `torch_inference.py` script https://github.com/openvinotoolkit/anomalib/blob/main/tools/inference/torch_inference.py. Maybe that will give you an idea on how to approach this. You can also have a look at this method https://github.com/openvinotoolkit/anomalib/blob/c82ddcd76bed47d196cffbee333a995e7f1df3ed/anomalib/deploy/inferencers/base_inferencer.py#L54
